src/data/collate.py

```diff
@@ -0,0 +1,47 @@
+"""Custom collate function for the data loader."""
+
+from typing import Any
+
+import torch
+from torch.nn.utils.rnn import pad_sequence
+
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+
+def custom_collate(batch: list[Any]) -> Any:
+    """
+    Custom collate function to be used in the data loader.
+    :param batch: list, with length equal to number of batches.
+    :return: processed batch of data [add padding to text, stack tensors in batch]
+    """
+    img, correct_capt, curr_class, wrong_capt, wrong_class, word_labels = zip(*batch)
```

Comment:
what's the reason behind `a, b, c = zip(*batch)` when you can just do `a, b, c = batch` ?

Reply:
batch is list of tuples, where list element is consisting of `[img, correct_capt, curr_clas... etc]`
we want to extract out all the images in that batch together and put it into `img` variable,
similarly extract all correct_capt present in batch and put it in variable `correct_capt`, that's why using `zip(*batch)`